deepspeed/runtime/zero/linear.py

```diff
@@ -33,8 +34,8 @@ def print_rank_0(message, debug=False, force=False):
 
 
 try:
-    autocast_custom_fwd = get_accelerator().amp().custom_fwd
-    autocast_custom_bwd = get_accelerator().amp().custom_bwd
+    autocast_custom_fwd = functools.partial(torch.amp.custom_fwd, device_type=get_accelerator().device_name())
```

Comment:
Thanks @terry-for-github - It looks like torch.amp is introduced in torch 1.11, @tjruwase thoughts on keeping support for older torch versions?

Reply:
@terry-for-github 

Could you add something like

```
if hasattr(torch, 'amp'):
    autocast_custom_fwd = functools.partial(torch.amp.custom_fwd, device_type=get_accelerator().device_name())
    autocast_custom_bwd = functools.partial(torch.amp.custom_bwd, device_type=get_accelerator().device_name())
else:
    autocast_custom_fwd = get_accelerator().amp().custom_fwd
    autocast_custom_bwd = get_accelerator().amp().custom_bwd
```

That way we can support older torch versions too (like 1.11)

Reply:
sure! I will try it.

Reply:
torch.amp.custom_f/bwd only exist in torch 2.4 even though torch.amp is introduced after torch 1.11(not included).

So what I understand is that the original code supports torch 1.11~2.3(or maybe older, I'm not sure) and my code only supports 2.4. What I need to do is fix my code so that it supports 1.11~2.4.

Am I right?

Reply:
See ab7ab617c1aec835a15759ea5bf90ec0aa7d7fad

Reply:
I think there is something wrong with the unit test.

Reply:
Yes, we've fixed the unit tests now, thanks!